executor/aggfuncs/func_sum.go

```diff
@@ -90,6 +90,16 @@ func (e *sum4Float64) UpdatePartialResult(sctx sessionctx.Context, rowsInGroup [
 	return nil
 }
 
+func (e *sum4Float64) MergePartialResult(sctx sessionctx.Context, src, dst PartialResult) error {
+	p1, p2 := (*partialResult4SumFloat64)(src), (*partialResult4SumFloat64)(dst)
+	if p1.isNull {
+		return nil
+	}
+	p2.val += p1.val
+	p2.isNull = false
```

Comment:
we can only safely modify `p2.val` when it's not null:
```go
if p2.isNull {
	*p2 = *p1
} else {
	p2.val += p1.val
}
```

Reply:
Can `p2.val` be guaranteed to '0' when it's NULL?

Reply:
For `sum4Float64 `, sure.